network/network.go

```diff
@@ -251,7 +255,9 @@ func (n *Network) Start() error {
 		return err
 	}
 	for _, prot := range n.protocols {
-		prot.Start()
+		if err = prot.Start(); err != nil {
+			return err
+		}
```

Comment:
I changed this to returning an error but I think we should revert it as it should just retry indefinitely

Reply:
I don't understand. This error is for starting the entire protocol....